File: app/src/main/java/de/bitb/astroskop/ui/main/circumstances/details/CircumstanceDetailsPresenter.java
package de.bitb.astroskop.ui.main.circumstances.details;

import javax.inject.Inject;

import de.bitb.astroskop.AstroApplication;
import de.bitb.astroskop.controller.CircumstanceController;
import de.bitb.astroskop.injection.IInjection;
import de.bitb.astroskop.injection.components.AppComponent;
import de.bitb.astroskop.model.Circumstance;
import de.bitb.astroskop.ui.base.BasePresenter;

public class CircumstanceDetailsPresenter extends BasePresenter<ICircumstancesDetailView> implements IInjection {

    @Inject
    protected CircumstanceController circumstanceController;

    private Circumstance circumstance;

    public CircumstanceDetailsPresenter(AstroApplication application, ICircumstancesDetailView view) {
        super(application, view);
    }

    @Override
    public void inject(AppComponent appComponent) {
        appComponent.inject(this);
    }

    public void onCreate(String circumstanceUuid) {
        circumstance = circumstanceController.getByUUID(circumstanceUuid);
        getView().initZodiacInfo(circumstance.getZodiac());
        getView().initHouseInfo(circumstance.getHouse());
        getView().initRulerInfo(circumstance.getRuler());

    }

}

File: app/src/main/java/de/bitb/astroskop/ui/main/circumstances/details/CircumstanceDetailsPresenter.kt
package de.bitb.astroskop.ui.main.circumstances.details

import javax.inject.Inject

import de.bitb.astroskop.BaseApplication
import de.bitb.astroskop.controller.CircumstanceController
import de.bitb.astroskop.injection.IInjection
import de.bitb.astroskop.injection.components.AppComponent
import de.bitb.astroskop.model.Circumstance
import de.bitb.astroskop.ui.base.BasePresenter

class CircumstanceDetailsPresenter(application: BaseApplication, view: ICircumstancesDetailView) : BasePresenter<ICircumstancesDetailView>(application, view), IInjection {

    @Inject
    var circumstanceController: CircumstanceController? = null

    private var circumstance: Circumstance? = null

    override fun inject(appComponent: AppComponent?) {
        appComponent?.inject(this)
    }

    fun onCreate(circumstanceUuid: String) {
        circumstance = circumstanceController!!.getByUUID(circumstanceUuid)
        getView().initZodiacInfo(circumstance!!.zodiac!!)
        getView().initHouseInfo(circumstance!!.house!!)
        getView().initRulerInfo(circumstance!!.ruler!!)
    }

}
